test/functional/test_framework/test_node.py

```diff
@@ -141,6 +160,33 @@ def node_encrypt_wallet(self, passphrase):
         self.rpc = None
         self.rpc_connected = False
 
+    def add_p2p_connection(self, p2p_conn_type, dstaddr='127.0.0.1', dstport=None, services=NODE_NETWORK, send_version=True):
```

Comment:
It seems very common to refer to `self.nodes[i].p2p`. Maybe this method should return the new connection so the client code can assign to a local variable.

Reply:
Good idea. I'll add that